fem/gridfunc.cpp

```diff
@@ -2393,29 +2393,39 @@ void GridFunction::ProjectCoefficient(Coefficient &coeff)
       }
       else
       {
-         // Define and assemble linear form
```

Comment:
This replaces the global L2 projection with local interpolation, in the NURBS case. There is a comment for this function in gridfunc.hpp explaining that global L2 projection is used in the NURBS case. That comment should be updated. Should we also keep the global L2 projection in a new function, e.g. `ProjectCoefficientGlobalL2`, which could be called for any space (not just NURBS)?

Reply:
Should there be a parallel version in `ParGridFunction`, with `ProjectCoefficientGlobalL2` being a virtual function? It looks like in parallel, the `GridFunction:: ProjectCoefficientGlobalL2` will be called, which assembles serial systems on each MPI rank. This is an L2 projection on each MPI rank's subdomain.